tools/tdbg/adminCommands.go

```diff
@@ -0,0 +1,744 @@
+	if err != nil {
+		return nil, fmt.Errorf("connect to Cassandra failed: %s", err)
+	}
+	return session, nil
+}
+
+func newESClient(c *cli.Context) (esclient.CLIClient, error) {
+	esUrl := c.String(FlagElasticsearchURL)
+	parsedESUrl, err := url.Parse(esUrl)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to parse URL: %s", err)
+	}
+
+	esConfig := &esclient.Config{
+		URL:      *parsedESUrl,
+		Username: c.String(FlagElasticsearchUsername),
+		Password: c.String(FlagElasticsearchPassword),
+	}
+
+	if c.IsSet(FlagVersion) {
+		esConfig.Version = c.String(FlagVersion)
+	}
+
+	client, err := esclient.NewCLIClient(esConfig, log.NewCLILogger())
+	if err != nil {
+		return nil, fmt.Errorf("Unable to create Elasticsearch client: %s", err)
+	}
+
+	return client, nil
+}
+
+// AdminGetNamespaceIDOrName map namespace
+func AdminGetNamespaceIDOrName(c *cli.Context) error {
+	namespaceID := c.String(FlagNamespaceID)
+	namespace := c.String(FlagNamespace)
+	if len(namespaceID) == 0 && len(namespace) == 0 {
+		return fmt.Errorf("Need either namespace or namespaceId")
+	}
+
+	session, err := connectToCassandra(c)
+	if err != nil {
+		return err
+	}
+
+	if len(namespaceID) > 0 {
+		tmpl := "select namespace from namespaces where id = ? "
+		query := session.Query(tmpl, namespaceID)
+		res, err := readOneRow(query)
+		if err != nil {
+			return fmt.Errorf("readOneRow: %s", err)
+		}
+		namespaceName := res["name"].(string)
+		fmt.Printf("namespace for namespaceId %v is %v \n", namespaceID, namespaceName)
+	} else {
+		tmpl := "select namespace from namespaces_by_name where name = ?"
+		tmplV2 := "select namespace from namespaces where namespaces_partition=0 and name = ?"
+
+		query := session.Query(tmpl, namespace)
+		res, err := readOneRow(query)
+		if err != nil {
+			fmt.Printf("v1 return error: %v , trying v2...\n", err)
+
+			query := session.Query(tmplV2, namespace)
+			res, err := readOneRow(query)
+			if err != nil {
+				return fmt.Errorf("readOneRow for v2: %s", err)
+			}
+			namespace := res["namespace"].(map[string]interface{})
+			namespaceID := gocql.UUIDToString(namespace["id"])
+			fmt.Printf("namespaceId for namespace %v is %v \n", namespace, namespaceID)
+		} else {
+			namespace := res["namespace"].(map[string]interface{})
+			namespaceID := gocql.UUIDToString(namespace["id"])
+			fmt.Printf("namespaceId for namespace %v is %v \n", namespace, namespaceID)
+		}
+	}
+	return nil
+}
+
+// AdminGetShardID get shardID
+func AdminGetShardID(c *cli.Context) error {
+	namespaceID := c.String(FlagNamespaceID)
+	wid := c.String(FlagWorkflowID)
+	numberOfShards := int32(c.Int(FlagNumberOfShards))
+
+	if numberOfShards <= 0 {
+		return fmt.Errorf("numberOfShards is required")
+	}
+	shardID := common.WorkflowIDToHistoryShard(namespaceID, wid, numberOfShards)
+	fmt.Printf("ShardId for namespace, workflowId: %v, %v is %v \n", namespaceID, wid, shardID)
+	return nil
+}
+
```

Comment:
There used to be a AdminDescribeTask which dumps the info for a transfer/timer/visibility/etc. task here.
Do we want to create a new admin API for it? 
I am ok with removing it actually, not seeing many use cases for it (task error logs already contains all necessary information). 
But want to confirm our decision here. cc @yiminc

Reply:
I think it could be useful, but can be added later.